From: Peter Green <plugwash@p10link.net>
Date: Wed, 23 Nov 2011 00:00:00 +0000
Subject: format string security

Fix FTBFS when building with hardening flags as used by dpkg 1.16.0.
Note that this was a *false positive* by the compiler: there was
in fact no security issue.  However we will work around it anyway.

Bug-Debian: http://bugs.debian.org/643461
---
 src/libqhull/io.c       | 2 +-
 src/testqset/testqset.c | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/libqhull/io.c b/src/libqhull/io.c
index 6601ffd..fb1a7fd 100644
--- a/src/libqhull/io.c
+++ b/src/libqhull/io.c
@@ -2238,7 +2238,7 @@ void qh_printfacet3math(FILE *fp, facetT *facet, qh_PRINT format, int notfirst)
     qh_memfree(point, qh normal_size);
   qh_settempfree(&points);
   qh_settempfree(&vertices);
-  qh_fprintf(fp, 9110, endfmt);
+  qh_fprintf(fp, 9110, "%s", endfmt);
 } /* printfacet3math */
 
 
diff --git a/src/testqset/testqset.c b/src/testqset/testqset.c
index 2bdcf17..309d3bf 100644
--- a/src/testqset/testqset.c
+++ b/src/testqset/testqset.c
@@ -214,7 +214,7 @@ void readOptions(int argc, char **argv, const char *promptstr, int *numInts, int
     char *endp;
 
     if (argc != 2 && argc != 3) {
-        printf(promptstr);
+        printf("%s", promptstr);
         exit(0);
     }
     numIntsArg= strtol(argv[1], &endp, 10);
